src/VecSim/vec_sim.cpp

```diff
@@ -8,6 +9,16 @@
 #include <cassert>
 #include "memory.h"
 
+timeoutCallbackFunction VecSimIndex::timeoutCallback = [](void *ctx) { return 0; };
+
+void VecSimIndex::setTimeoutCallbackFunction(timeoutCallbackFunction callback) {
```

Comment:
not a part of vecsimIndex?

Reply:
It is, but there is no `vec_sim_index.cpp` file, and if I try to define it in `vec_sim_index.h` I get an error because it is abstract and the function is defined twice (in bf and in hnsw)

Reply:
so let's have `vec_sim_index.cpp`